scipy/stats/_qmc.py

```diff
@@ -0,0 +1,1493 @@
+import math
+import warnings
+
+import numpy as np
+
+from scipy.optimize import brute
+from scipy._lib._util import check_random_state
+from scipy.optimize import basinhopping
+from scipy.stats import norm
+from scipy.stats._sobol import (
+    initialize_v, _cscramble, _fill_p_cumulative, _draw, _fast_forward,
+    _categorize, initialize_direction_numbers, _MAXDIM, _MAXBIT
+)
+
+__all__ = ['scale', 'discrepancy', 'QMCEngine', 'Sobol', 'Halton',
+           'OrthogonalLatinHypercube', 'LatinHypercube', 'OptimalDesign',
+           'multinomial_qmc', 'NormalQMC', 'MultivariateNormalQMC']
+
+
+def scale(sample, bounds, reverse=False):
+    """Sample scaling from unit hypercube to bounds range.
+
+    To convert a sample from :math:`[0, 1)` to :math:`[a, b), b>a`, the
+    following transformation is used:
+
+    .. math::
+
+        (b - a) * sample + a
+
+    Parameters
+    ----------
+    sample : array_like (n_samples, dim)
+        Sample to scale.
+    bounds : tuple or array_like ([min, dim], [max, dim])
+        Desired range of transformed data. The transformation apply the bounds
+        on the sample and not the theoretical space, unit cube. Thus min and
+        max values of the sample will coincide with the bounds.
+    reverse : bool
+        Reverse the transformation, from bounds range to unit hypercube.
+
+    Returns
+    -------
+    sample : array_like (n_samples, dim)
+        Scaled-sample.
+
+    Examples
+    --------
+    >>> from scipy.stats import qmc
+    >>> bounds = [[-2, 0],
+    ...           [6, 5]]
+    >>> sample = [[0.5 , 0.5 ],
+    ...           [0.75, 0.25]]
+    >>> qmc.scale(sample, bounds)
+    array([[2.  , 2.5 ],
+           [4.  , 1.25]])
+
+    """
+    bounds = np.asarray(bounds)
+    min_ = np.min(bounds, axis=0)
+    max_ = np.max(bounds, axis=0)
+    if not reverse:
+        return sample * (max_ - min_) + min_
+    else:
+        return (sample - min_) / (max_ - min_)
+
+
+def discrepancy(sample, iterative=False, method='CD'):
+    """Discrepancy on a given sample.
+
+    Parameters
+    ----------
+    sample : array_like (n_samples, dim)
+        The sample to compute the discrepancy from.
+    iterative : bool
+        Must be False if not using it for updating the discrepancy.
+    method : str
+        Type of discrepancy, can be ['CD', 'WD', 'MD', 'star'].
+
+    Returns
+    -------
+    discrepancy : float
+        Centered discrepancy.
+
+    Notes
+    -----
+    The discrepancy is a uniformity criterion used to assess the space filling
+    of a number of samples in a hypercube.
+    The discrepancy measures how the spread of the points deviates from a
+    uniform distribution.
+    The lower the value is, the better the coverage of the parameter space is.
+
+    Taking a subspace of the parameter space we count the number of points in
+    the subpace and compare it to the total number of the points of the
+    sample. This value is substracted by the volume of the subspace. This
```

Comment:
```suggestion
    sample. This value is substrated by the volume of the subspace. This
```
Does this mean the volume of the subspace is subtracted _from_ the number of points, or is it the other direction (the number of points is subtracted _from_ the volume of the subspace)? Or something else? It seems that there needs to be some sort of normalization to compare a number of points to a volume in a unit hypercube.

Reply:
Could you clarify this in the documentation? I think it is likely that other readers would have the same question.